### PR TITLE
Interactive branch

### PR DESCRIPTION
**Pull request by jonas-45**
The interactive branch has the following features implemented in the project;
A change event listener is added to checkboxes to mark tasks as completed
A clear-all-completed button is implemented to remove all completed tasks using the array filter method
To-do list tasks are persisted in local storage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,18 +24,28 @@ taskDescriptionInput.addEventListener('keypress', (e) => {
...
+//     }
+//   });
+
+//   uncompleted.forEach((task, index) => {
+//     task.index = index + 1;
+//   });
+//   taskObj.tasksArr = uncompleted;
+//   taskObj.updateLocalStorage();
+//   window.location.reload();
+// });
```
Comment (muneebulrehman) [inline src/index.js]: Kindly remove all the commented code to make your code base clean.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @jonas-45 , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

- [ ] When I try to delete a task, it makes the application go into an infinite loop. I have recorded [this video](https://www.loom.com/share/79b51898b1674763b02e16bbb733efa9) where I demonstrated the problem in detail. Kindly fix this error then submit for a re-review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @jonas-45, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (jonas-45) [reply src/index.js]: Thanks. I will work on it right away

**Comment by jonas-45** [comment]
Thank you for your time and effort in reviewing my work

**Pull request merged**